kensu/google/cloud/bigquery/job/bigquery_stats.py

```diff
@@ -37,25 +37,26 @@ def compute_bigquery_stats(table_ref=None, table=None, client=None, stats_aggs=N
     selector = ",".join([sql_aggregation + " " + col.replace(".","__ksu__") + "_" + stat_name
                          for col, stats_for_col in stats_aggs.items()
                          for stat_name, sql_aggregation in stats_for_col.items()])
+
+    for nested in sorted(unnest,key=len,reverse=True):
+        if "." in nested:
+            selector=selector.replace(nested,nested.replace(".","__ksu__"))
+
     filters = ''
-    unnest = ''
     if input_filters is not None and len(input_filters) > 0:
         filters = f"WHERE {' AND '.join(input_filters)}"
-    if list_of_unnested != []:
-        unnest =[]
-        for el in list_of_unnested:
-            unnest.append(f",UNNEST({el}) AS {el}")
-        unnest = "".join(unnest)
+    if unnest != []:
+        list_unnest =[]
+        for el in unnest:
+            list_unnest.append(f",UNNEST({el}) AS {el.replace('.','__ksu__')}")
+    unnest = "".join(list_unnest) if unnest != [] else ''
```

Comment:
a spec or even better `doctest` would help a lot to grasp what's happening, e.g. doctest could be for extracted sub function to make it simpler